### DISCUSSION TITLE
PBA@UCB Content Dry Runs

### DISCUSSION THREAD

**Discussion by nuke-web3** [Polkadot-Blockchain-Academy/pba-content#603] [General]
In order to ensure that we always test things before our students are exposed to it, PBA is committed to beta testing and dry runs of content with peer review and feedback that is used to refine our content and finally deliver it to our students.

Some factors that signal a beta test and/or dry run are needed:
- A new faculty member who will be instructing for the first time with us on established content
- New content is added to any module (lecture slides, exercises, assignments, etc.)
- Existing content is majorly refactored  (lecture slides, exercises, assignments, etc.)

Dry runs for live-presented content will happen _remotely_ in the form of a  (recorded) video call sessions with an open invitation to (soon to be on-boarded) beta testers from the PBA Alumni, Parity and W3F.

Dry runs will be limited to a single lesson in duration (~1 hour is a good limit if possible) with anther 25% time buffer added for synchronous feedback from participants. Other than the presenter present, we will need a scribe from PBA to take notes and outline takeaways from the dry runs.

### Pre-PBA@UCB dry runs

The schedule for these is TBD by availability of those who need to present (some are outlined below, please call out more needed), but will need to occur within the time frame of **mid May to the end of June 2023** so we conclude all dry run sessions well before PBA@UCB kicks off July 10th.

Here is a listing of the modules & specific content I believe need a dry-run:

- Cryptography with @naterarmstrong  & @NukeManDan 
- Contracts with @laurenelee  (& @aaronbassett & @JoshOrndorff & maybe more? if they are also presenting)
- FRAME with @shawntabrizi & @aaronbassett  (Primarily for any lessons Aaron will take on) 
- Polkadot with @shawntabrizi  & @BradleyOlson64  (Although these will likely happen in person at the Lisbon retreat in June)
- XCM with Gav and @KiChjang  (To happen at the Lisbon retreat)
- New content (#610 ) by @pepyakin 
- New content in (#609) by @bhargavbh 

---

### Action items

1. Am I missing anything in the above list we think we _need_ a dry run or beta test of? Please call it out here!
1. For those that do need/want a dry run scheduled, please enumerate what lessons and specifically and estimate the time needed to deliver the content & feedback session.

**Comment by shawntabrizi** [comment]
@aaronbassett what kinds of lessons do you feel that you could teach?

I'll be a little honest, I have some skepticism that from learning FRAME the first time at the academy to teaching it at the next one seems like a fast progression to me. But I guess it can depend on the content, and mastery of that material.

**Reply by aaronbassett** [reply]
@shawntabrizi I completely understand your scepticism, and I wouldn't feel comfortable taking responsibility for the module's content at the moment either, especially not at such short notice.

No one will know the content as intimately as the core engineers. Even if I were to focus all of my time on becoming fully versed in the entirety of FRAME—as it stands today—my knowledge would be outdated as soon as I finished; its development progresses so quickly.

But I am experienced in Pedagogy, Andragogy, and Heutagogy. I can add the most value by partnering with a subject matter expert to help plan and structure the module, prepare lessons, and even deliver the prepared content on the day. 

As my experience with the material increases, I can assume more ownership of the module, with less reliance upon the subject matter expert. However, I would never develop the lessons in isolation. It is imperative that core engineering sets the direction and provides feedback and oversight.

---

That said, since PBA, there are several areas of FRAME I've worked with extensively and already feel very comfortable with. 

These lessons I know in-depth and could present with minimal preparation:

- FRAME Storage
- Events and Errors
- FRAME Calls

I know these lessons well, but I would need some time to familiarise myself with particular parts:

- Introduction To FRAME
- FRAME Tips and Tricks
- Pallet Coupling
- Pallet Hooks
- Origins

I have some proficiency in the content of these lessons, but I would need support in reaching a level of understanding where I would feel comfortable teaching them:

- Construct Runtime and Tests
- FRAME Pallets and Traits
- Outer Enum

For the remainder of the lessons, which account for most of 5.4 and 5.5, I have not had much opportunity to revisit them since PBA, so I need to spend considerable time learning those again before I consider myself capable of teaching others.

**Comment by Sophia-Gold** [comment]
I will need to do a dry run and won't be in Lisbon. I'm developing and presenting one topic broken into two lectures. I would plan for two hours total.

**Comment by nuke-web3** [comment]
I have crafted (with the support of @Sam-Srey-Parity   🦸) a new **github project** that will help us track the progress of **key content delivery related milestones** as we march towards shipping content at Berkeley together:

### https://github.com/orgs/Polkadot-Blockchain-Academy/projects/4/views/1

This will get more issues/PRs added to it as we progress for those looking for a holistic view of content dev progress. To start, I have added a tracking issue per module that **all module leads/instructors/TAs will need to update with concrete plans on dry runs and beta testing** so we can ensure that we are on track to sign-off on content readiness before July 10th for all content.

Please do update those [open dry run issues](https://github.com/Polkadot-Blockchain-Academy/pba-content/issues?q=is%3Aissue+is%3Aopen+Dry+Runs+%26+Beta+Testing) so we can get specific dates locked in and invitations for Alumni and Parity people to attend these dry runs and provide feedback ASAP 😁
